### PR TITLE
Adds a basic historydata.php API function to grab history.md content.

### PR DESCRIPTION
**Pull request by jahorton**
Please let me know if it's not the right approach - I figured I'd go for something relatively basic with this.

Designed to be accessed via downloads.keyman.com/api/history/[platform], providing a raw dump of the platform's markdown.

---

The expectation of this API - nightly rsyncs will refresh history.md with the latest history data via `rsync` call in the platform's base folder.  (The master branch's history.md should always serve as the 'master' version.)

### CODE REVIEW

File: api/historydata.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+// Doing the actual work.
+
+function get_history_contents($platform) {
+// Reads straight from a file.  Likely to be useful for the history-exposing API.
+  $inputURL = "../$platform/history.md";
+
+  $contents = @file_get_contents($inputURL);
+
+  if($contents == FALSE) {
```
Comment (mcdurdin) [inline api/historydata.php]: Should be `===` for type safety and avoidance of coercion but probably doesn't matter in reality.

File: api/historydata.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    fail("Cannot locate history information!");
+  }
+  echo $contents;
+}
+
+for($i=0; $i < count($platforms); $i++) {
```
Comment (mcdurdin) [inline api/historydata.php]: I don't see a real need to support dumping history from all platforms at once. One at a time is fine.

**Review by mcdurdin — CHANGES_REQUESTED** [review]
Minor changes requested.

**Comment by mcdurdin** [comment]
So seems to me CI nightly for each platform should publish latest history.md to the top-level folder for that platform. Easy enough to accomplish.

File: api/historydata.php (hunk 2 above)
Comment (jahorton) [reply api/historydata.php]: So, it should return an error if a platform isn't specified, then?  I figured it should probably have *some* kind of default behavior, like with versioninfo.php, though this default does seem admittedly less useful.

Comment (mcdurdin) [reply api/historydata.php]: Yes, just return an error if the platform isn't specified. It's an internal API so we don't have to be too friendly in error cases.

**Review by mcdurdin — APPROVED** [review]

**Pull request merged**